qa/tasks/cephfs/test_quota.py

```diff
@@ -104,3 +104,52 @@ def test_remote_update_write(self):
         with self.assertRaises(CommandFailedError):
             self.mount_b.write_n_mb("subdir_data/file", 40)
 
+    def test_human_readable_quota_values(self):
+        self.mount_a.run_shell(["mkdir", "subdir"])
+
+        self.assertEqual(self.mount_a.getfattr("./subdir",
+                                               "ceph.quota.max_bytes"), None)
+
+        # kibibyte
+        self.mount_a.setfattr("./subdir", "ceph.quota.max_bytes", "10k")
+        self.assertEqual(self.mount_a.getfattr(
+            "./subdir", "ceph.quota.max_bytes"), "10240")
+        self.mount_a.setfattr("./subdir", "ceph.quota.max_bytes", "100K")
+        self.assertEqual(self.mount_a.getfattr(
+            "./subdir", "ceph.quota.max_bytes"), "102400")
+
+        # mebibyte
+        self.mount_a.setfattr("./subdir", "ceph.quota.max_bytes", "10m")
+        self.assertEqual(self.mount_a.getfattr(
+            "./subdir", "ceph.quota.max_bytes"), "10485760")
+        self.mount_a.setfattr("./subdir", "ceph.quota.max_bytes", "100M")
+        self.assertEqual(self.mount_a.getfattr(
+            "./subdir", "ceph.quota.max_bytes"), "104857600")
+
+        # gibibyte
+        self.mount_a.setfattr("./subdir", "ceph.quota.max_bytes", "2g")
+        self.assertEqual(self.mount_a.getfattr(
+            "./subdir", "ceph.quota.max_bytes"), "2147483648")
+        self.mount_a.setfattr("./subdir", "ceph.quota.max_bytes", "4G")
+        self.assertEqual(self.mount_a.getfattr(
+            "./subdir", "ceph.quota.max_bytes"), "4294967296")
+
+        # tebibyte
+        self.mount_a.setfattr("./subdir", "ceph.quota.max_bytes", "1t")
+        self.assertEqual(self.mount_a.getfattr(
+            "./subdir", "ceph.quota.max_bytes"), "1099511627776")
+        self.mount_a.setfattr("./subdir", "ceph.quota.max_bytes", "2T")
+        self.assertEqual(self.mount_a.getfattr(
+            "./subdir", "ceph.quota.max_bytes"), "2199023255552")
+
+        # invalid arguments
+        with self.assertRaises(CommandFailedError):
+            self.mount_a.setfattr("./subdir", "ceph.quota.max_bytes", "10A")
+        with self.assertRaises(CommandFailedError):
+            self.mount_a.setfattr("./subdir", "ceph.quota.max_bytes", "1y00K")
+        with self.assertRaises(CommandFailedError):
+            self.mount_a.setfattr("./subdir", "ceph.quota.max_bytes", "af00")
+        with self.assertRaises(CommandFailedError):
+            self.mount_a.setfattr("./subdir", "ceph.quota.max_bytes", "G")
+        with self.assertRaises(CommandFailedError):
+            self.mount_a.setfattr("./subdir", "ceph.quota.max_bytes", "")
```

Comment:
I know string parsing is painful, but here are some more suggestions for size spec testing:
- Just a space character, e.g. " "
- a negative number with suffix, e.g. "-1t"
- just a negative number, e.g. "-1"
- zero with suffix, e.g. "0k"
- just zero, e.g. "0"

although most of these values will be valid in the lexical_cast<> operation, you'll also need to address these unacceptable values in your error logging appropriately

Reply:
just `0` is used to remove quota. it isnt an invalid value right

Reply:
yeah; the docs do say so

Reply:
done.